viadot/sources/spectrum.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Dict, Literal
+import boto3
+
+import pandas as pd
+import awswrangler as wr
+
+
+class Spectrum:
+    """
+    A class for pulling data from and uploading to Spectrum.
+
+    Args:
+        profile_name (str, optional): The name of the profile.
+        aws_secret_access_key (str, optional): AWS secret access key
+        aws_session_token (str, optional): AWS temporary session token
+    """
+
+    def __init__(
+        self,
+        profile_name: str = None,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+    ):
+        if profile_name:
+            self.session = boto3.session.Session(profile_name=profile_name)
+        elif aws_access_key_id and aws_secret_access_key:
+            self.session = boto3.session.Session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+            )
+        else:
+            self.session = boto3.session.Session()
+
+    def ls(
+        self, database: str = None, name_contains: str = None, search_text: str = None
+    ) -> List[str]:
+        """
+        Returns a list of tables in a Glue database.
+
+        Args:
+            database (str, optional): Database name.
+            name_contains (str, optional): Select by a specific string on table name.
+            search_text (str, optional): Select only tables with the given string in table's properties.
+        """
+        df = wr.catalog.tables(
+            boto3_session=self.session,
+            database=database,
+            name_contains=name_contains,
+            search_text=search_text,
+        )
+
+        return df
+
+    def exists(self, database: str, table: str) -> bool:
+        """
+        Check if a table exists in Glue database.
+
+        Args:
+            database (str): AWS Glue catalog database name.
+            table (str): AWS Glue catalog table name.
+
+        Returns:
+            bool: Whether the paths exists.
+        """
+        return wr.s3.does_table_exist(
```

Comment:
Is this correct that it's `wr.s3`? Other methods use `wr.catalog`, hence my confusion

Reply:
Agreed. It should use `wr.catalog`. 
Fixed in the [🐛 fixed bug](https://github.com/dyvenia/viadot/pull/579/commits/60a314434d68ab2fdcc2ac31960a1fcdbc174d51)